File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        int arr[201] = {0};
        int maxelement = 0;

        for (int i = 0; i < nums.size(); i++) {
            arr[nums[i]]++;
            maxelement = max(maxelement, arr[nums[i]]);
        }

        vector<vector<int>> res;
        int i = 1;

        while (i <= maxelement) {
            vector<int> list;
            for (int j = 0; j < 201; j++) {
                if (arr[j] >= i) {
                    list.push_back(j);
                }
            }
            if (!list.empty()) {
                res.push_back(list);
            }
            i++;
        }
        return res;
    }
};

File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        result = []
        maxElement, i = 0, 1
        count = [0 for i in range(201)]
        for k in range(len(nums)):
            count[nums[k]] += 1
            maxElement = max(maxElement, count[nums[k]])
        
        while i <= maxElement:
            temp = []
            for j in range(201):
                if count[j] >= i:
                    temp.append(j)
            if temp:
                result.append(temp)
            
            i += 1
        return result